misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/manufacturing/RawMaterialIssueScanScreen.jsx

```diff
@@ -7,92 +7,113 @@ import counterpart from 'counterpart';
 
 import { selectWFProcessFromState } from '../../../reducers/wfProcesses_status';
 import { updateManufacturingIssueQty, updateManufacturingIssue } from '../../../actions/ManufacturingActions';
+import { pushHeaderEntry } from '../../../actions/HeaderActions';
 import { toastError } from '../../../utils/toast';
+import { getLocation } from '../../../utils';
 
 import StepScanScreenComponent from '../common/StepScanScreenComponent';
 
-function RawMaterialIssueScanScreen(WrappedComponent) {
-  const mapStateToProps = (state, { match }) => {
-    const { workflowId: wfProcessId, activityId, lineId, stepId, appId } = match.params;
+class RawMaterialIssueScanScreen extends PureComponent {
+  constructor(props) {
+    super(props);
 
-    const wfProcess = selectWFProcessFromState(state, wfProcessId);
-    const stepProps = wfProcess.activities[activityId].dataStored.lines[lineId].steps[stepId];
+    this.state = {
+      scannedBarcode: null,
+    };
+  }
+
+  componentDidMount() {
+    const {
+      stepProps: { huBarcode, qtyToIssue },
+    } = this.props;
+    const location = getLocation(this.props, true);
```

Comment:
use only the required params

Reply:
tbh I see no real value in doing this, but I just want this to be merged so I'll comply. We create 3 temp variables

```
const { wfProcessId, activityId, lineId } = this.props;
```

and then another object.

```
manufacturingLineScreenLocation({ wfProcessId, activityId, lineId });
```

when passing all props doesn't really cost us anything since it's just a reference.

Reply:
I was also thinking about that.
I don't have strong arguments for this and I am not 100% how is better, so feel free to suggest... i.e. I can go with your suggestion if you think it's better.

My thinking was that if we decompose and then send only those 3 params we will gain the following:
* eslint advice in case one of the props are missing
* code easier to read and to understand what's doing on and how the data is flowing
* IMHO the performance for one approach or the other is the same